src/Build/Graph/ProjectGraph.cs

```diff
@@ -169,17 +172,19 @@ public ProjectGraph(IEnumerable<ProjectGraphEntryPoint> entryPoints, ProjectColl
         {
             ErrorUtilities.VerifyThrowArgumentNull(projectCollection, nameof(projectCollection));
 
-            var nodeStates = new Dictionary<ConfigurationMetadata, NodeState>();
+            var nodeStates = new Dictionary<ProjectGraphNode, NodeState>();
             var entryPointNodes = new List<ProjectGraphNode>();
+            var tasksInProgress = new ConcurrentDictionary<ConfigurationMetadata, Task>();
             foreach (var entryPoint in entryPoints)
             {
                 PropertyDictionary<ProjectPropertyInstance> globalPropertyDictionary = CreatePropertyDictionary(entryPoint.GlobalProperties);
                 var configurationMetadata = new ConfigurationMetadata(FileUtilities.NormalizePath(entryPoint.ProjectFile), globalPropertyDictionary);
-                if(!nodeStates.TryGetValue(configurationMetadata, out NodeState nodeState))
+                LoadGraph(configurationMetadata, projectCollection, globalPropertyDictionary, tasksInProgress);
+                entryPointNodes.Add(_allParsedProjects[configurationMetadata]);
+                if (!nodeStates.TryGetValue(_allParsedProjects[configurationMetadata], out NodeState nodeState))
```

Comment:
Several of these dictionary accesses have a throwaway value. C# now allows a nice pattern for that with the `_` variable.

```suggestion
                if (!nodeStates.TryGetValue(_allParsedProjects[configurationMetadata], out var _))
```